webpack.config.js

```diff
@@ -0,0 +1,71 @@
+const webpack = require("webpack");
+const path = require("path");
+const HtmlWebpackPlugin = require("html-webpack-plugin");
+const PnpWebpackPlugin = require("pnp-webpack-plugin");
+// Not used directly in playground, because it imports typescript
+const { SyntaxKind: LuaSyntaxKind } = require("typescript-to-lua/dist/LuaAST");
+
+const resolve = query => path.resolve(__dirname, query);
+
+/** @type {import("webpack").Configuration} */
+module.exports = {
+    devtool: "source-map",
+    entry: {
+        landing_bundle: resolve("src/landing/index.ts"),
+        play_bundle: resolve("src/playground/index.ts"),
+    },
+    output: { path: resolve("dist") },
+    node: { fs: "empty", module: "empty" },
+    resolve: {
+        extensions: [".tsx", ".ts", ".js"],
+        plugins: [PnpWebpackPlugin],
```

Comment:
As far as I can tell this is required because of yarn 2?

Is there any benefit for using yarn 2 instead of npm? Does it enable any features that weren't available before?
Because right now the only thing that I can see is that we need an additional dependence, which always is a bad thing imo.

Im not really in the loop but I heard there is a bit of controversy around yarn 2, the decision to use prolog, slow installs on a cold cache etc.. and some other technical choices.

Reply:
> As far as I can tell this is required because of yarn 2?

Yes (though webpack 5 would support pnp natively)

> Is there any benefit for using yarn 2 instead of npm?

(I think that could be quite a long discussion, so maybe let's do it in Discord?)

Reply:
Im sure this can be a long discussion. But since im not really familiar with the internals of yarn or npm I doubt I could contribute something interesting :D

What I was going for with the above comment, was more along the lines:
Does yarn add any new (useful) high level feature that we did not have before? Or does is the only thing that changes that I have do download yet another JS package manager.

I just don't really see the benefit of introducing another dependency and using different package managers for different TypescriptToLua projects, regardless of which one is better.

Reply:
Yarn doesn't unlock something we couldn't do with npm, it's just about QoL features, like:
- faster installs (at least with warm cache)
- less bugs and a more modern codebase (Yarn is written in TypeScript)
- it has a plugin that auto-installs `@types` packages
- [`yarn upgrade-interactive`](https://yarnpkg.com/cli/upgrade-interactive)
- a single system-wide cache (TSTL's `node_modules` is 132M, which quickly adds up when you maintain many packages)

Reply:
Actually, looks like yarn doesn't recommend the global installation method anymore, recommending per-project installation, committing yarn binary to the repo, which I personally really don't like as a path forward. So I'm fine with reverting to the npm until the dust settles